services/management/dbaas/components_service.go

```diff
@@ -192,6 +194,93 @@ func (c componentsService) ChangePXCComponents(ctx context.Context, req *dbaasv1
 	return &dbaasv1beta1.ChangePXCComponentsResponse{}, nil
 }
 
+type installedComponentsVersion struct {
+	kuberentesClusterName string
+	pxcOperatorVersion    string
+	psmdbOperatorVersion  string
+}
+
+func (c componentsService) getInstalledOperatorsVersion(ctx context.Context, wg *sync.WaitGroup, responseCh chan installedComponentsVersion, kuberentesCluster *models.KubernetesCluster) {
+	defer wg.Done()
+	resp, err := c.dbaasClient.CheckKubernetesClusterConnection(ctx, kuberentesCluster.KubeConfig)
+	if err != nil {
+		c.l.Errorf("failed to check kubernetes cluster connection: %v", err)
+		return
+	}
+	responseCh <- installedComponentsVersion{
+		kuberentesClusterName: kuberentesCluster.KubernetesClusterName,
+		pxcOperatorVersion:    resp.Operators.Xtradb.Version,
+		psmdbOperatorVersion:  resp.Operators.Psmdb.Version,
+	}
+}
+
+func doesOperatorNeedUpdate(installedOperatorVersion, latestOperatorForInstalledPMM *goversion.Version) (availableOperatorVersion string) {
```

Comment:
why do we use a named return value here?

also when we use named return values, function has naked return, I would suggest we remove the named return

Reply:
I did it to make it more clear what is a return value.